airflow/models/baseoperator.py

```diff
@@ -238,40 +238,40 @@ class derived from this one results in the creation of a task object,
     @apply_defaults
     def __init__(
         self,
-        task_id,  # type: str
-        owner=configuration.conf.get('operators', 'DEFAULT_OWNER'),  # type: str
-        email=None,  # type: Optional[str]
-        email_on_retry=True,  # type: bool
-        email_on_failure=True,  # type: bool
-        retries=0,  # type: int
-        retry_delay=timedelta(seconds=300),  # type: timedelta
-        retry_exponential_backoff=False,  # type: bool
-        max_retry_delay=None,  # type: Optional[datetime]
-        start_date=None,  # type: Optional[datetime]
-        end_date=None,  # type: Optional[datetime]
+        task_id: str,
+        owner: str = configuration.conf.get('operators', 'DEFAULT_OWNER'),
+        email: Optional[str] = None,
```

Comment:
No need for Optional here - it will be inferred automatically via = None. (though maybe you want to leave it for clarity).

Reply:
My 2c, I am personally a big fan of `--no-implicit-optional` in mypy checks. I find that it forces contributors to reason explicitly about null safety, and is much clearer for readers.

Reply:
I agree it seems sensible to remove to optionals. However when reading the docs (https://docs.python.org/3/library/typing.html#typing.Optional), it says the Optional is appropriate:

> ... if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not. For example:
> 
> ```python
> def foo(arg: Optional[int] = None) -> None:
>     ...
> ```

Reply:
Yeah. Explicit is better than implicit.